testnet/src/tests/integrations.rs

```diff
@@ -1,102 +1,24 @@
-use vm::{
-    database::{ HeadersDB, ClaritySerializable },
-    types::{QualifiedContractIdentifier, TupleData, PrincipalData},
-    analysis::{mem_type_check, contract_interface_builder::{build_contract_interface, ContractInterface}},
-    clarity::ClarityConnection,
-    Value, ClarityName, ContractName, errors::RuntimeErrorType, errors::Error as ClarityError };
-use chainstate::stacks::{
-    db::StacksChainState, C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
-    StacksMicroblockHeader, StacksPrivateKey, TransactionSpendingCondition, TransactionAuth, TransactionVersion,
-    StacksPublicKey, TransactionPayload, StacksTransactionSigner,
-    TokenTransferMemo, CoinbasePayload, TransactionPostConditionMode,
-    StacksTransaction, TransactionSmartContract, TransactionContractCall, StacksAddress };
-use chainstate::burn::VRFSeed;
-use burnchains::Address;
-use address::AddressHashMode;
-use net::{Error as NetError, StacksMessageCodec, AccountEntryResponse, ContractSrcResponse, CallReadOnlyRequestBody};
-use util::{log, strings::StacksString, hash::hex_bytes, hash::to_hex};
 use std::collections::HashMap;
-use util::db::{DBConn, FromRow};
 
-use std::{thread, time};
+use stacks::vm::{
```

Comment:
Will let @kantai speak more to how appropriate this is, but I thought these integration tests were _not_ part of the testnet?  I thought they were specifically meant to test interactions between aspects of the Clarity VM?  If so, then they should be kept in `src/vm`.

Reply:
I only moved the tests relying on the testnet frameworks. They are not part of the testnet, but they depend on this version of the testnet. If we're saying that the testnet is a separate module with its own dependencies, if we move this tests back to their initial location `src/vm`, we're introducing a dependency cycle (tesnet depends on vm, vm depends on testnet).

Reply:
In general, I prefer that re-orging code be separate from PRs that implement new features. It can make it hard to evaluate what's changed in a PR, and can make merges pretty painful.

In this case, these tests are testing a few different aspects of the clarity vm when integrated with other components (namely chainstate and the rpc) which requires testing with some kind of instrumented node (in this case, it was depending on helium). It's not specifically unit testing Clarity, so it's fine to move, and perhaps wise to move, if we at any point want to be able to build Clarity as a separate crate (may help solve our OOM problems during compilation, may help for building things like the language server or WASM targets).

Reply:
Sounds good.  I've generally been in favor of keeping all tests for a module in the same file, but if we see ourselves building out more helium-based integration tests, then this is fine by me.